mlflow/pipelines/step.py

```diff
@@ -233,6 +233,14 @@ def get_artifacts(self) -> List[Any]:
         """
         return {}
 
+    @experimental
+    @abc.abstractmethod
+    def is_predict_step(self) -> bool:
```

Comment:
I really don't like this approach. Why is `predict` so special? What if we want to add another subgraph, e.g. `deploy` with only one step `deploy`? Can we think about a more general approach to solve this issue, e.g. through some enum or predefined subgraphs?
A very rough thought, if we explicitly define subgraphs in the base class, e.g.
```
_TRAINING_STEPS = {"ingest", "split", "train", "evaluate", "register"}
_INFERENCE_STEPS = {"ingest_scoring", "predict"}
# (in the future) _DEPLOY_STEPS = {"deploy"}
```
then you may be able to tell which subgraph the step is in based on the predefined name sets.

Reply:
I contemplated using an enum. It seems like a premature optimization to me, but happy to do it! 

I don't think we should define subgraphs in the base class, that violates open-closed. Too easy to forget to update those constants when adding/removing steps.

Reply:
Agreed with both points here. I'd suggest we leave the structure as-is for now :)

Reply:
Say this is updated to:
```
def step_class(self) -> StepClass
```
The benefit of this approach is you don't have to duplicate code across pipelines implementations. Can you help me understand your opposition to this approach?

Reply:
`StepClass` reads much better. @dbczumar I have trouble understand your comment. Which "points" do you agree, and what does it mean by "leave the structure as-is"? What "structure" were you referring to?

Reply:
I was saying that I prefer leaving master as-is. Wasn't thrilled with special casing prediction and also wasn't thrilled with the idea of defining the pipeline subgraphs in the base class.

StepClass sounds good to me!

Reply:
Great, I agree this is in a better state now, so thank you for the feedback!